dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -97,11 +108,15 @@ inline void ParseIndexFile(std::vector<SampleDesc>& samples_container,
   std::string index_version;
   DALI_ENFORCE(global_meta_stream >> index_version,
                IndexFileErrMsg(index_path, 0, "no version signature found"));
-  DALI_ENFORCE(kCurrentIndexVersion == index_version,
-               IndexFileErrMsg(
-                   index_path, 0,
-                   "the version of the index file does not match the expected version (expected: ",
-                   kCurrentIndexVersion, " actual: ", index_version, ")"));
+  DALI_ENFORCE(
+      kCurrentIndexVersion == index_version ||
+          std::accumulate(
+              kLegacyIndexVersions.begin(), kLegacyIndexVersions.end(), false,
+              [&index_version](bool a, const std::string& b) { return a || index_version == b; }),
+      IndexFileErrMsg(
+          index_path, 0,
+          make_string("The version of the index file (", index_version,
+                      ") does not match either the latest or any of the legacy versions.")));
```

Comment:
I think this message is too verbose.
```suggestion
          make_string("Unsupported version of the index file (", index_version, ")")));
```

Reply:
Done